status-im/status-react
# Add unpinning of mailservers

### Pinned mailservers

Currently once a user select a mailserver, that mailserver will be "pinned". This means that we won't be automatically selecting a different mailserver in case there are some connection problems. We will instead show a pop-up asking the user to retry or change mailserver.

The issue currenty is that once you pin a mailserver, there's no way back, so you will be presented with the pop up everytime.

This PR adds a "Unpin" feature, that allows user to forget their preference, so that the pop up won't be displayed, and we'll rotate the changes.

### Testing

1) Disable automatic selection
2) Replicate failed requests to the mailserver (trying going offline/Fetch 24 hours, non existent mailserver)
3) You should see a pop up
4) Enable automatic selection
5) It should rotate automatically without asking you.

New accounts should have automatic selection enabled.

For now only implemented on mobile, will implement on desktop in a separate PR.

Known issues:
- When enabling automatic selection, it will change mailserver.

@errorists it looks ugly, I have added a couple of screenshots, i i's under advanced, but for sure can be improved upon, what do you think, is it ok for a rough cut or there's some magic to apply?

![disabled](https://user-images.githubusercontent.com/1017008/53638030-37d9fc00-3c25-11e9-8a31-ee3a53ae8df6.png)
![enabled](https://user-images.githubusercontent.com/1017008/53638031-37d9fc00-3c25-11e9-9fbb-49c77854efb0.png)



status: ready
